### PR TITLE
feat: support output CSVof transcribing

### PR DESCRIPTION
**Pull request by yanirmr**

### CODE REVIEW

**Comment by yoadsn** [comment]
LGTM

File: evaluate_model.py [hunk 1]
```diff
@@ -278,6 +304,12 @@ def evaluate_model(transcribe_fn, ds, text_column):
         ds = datasets.load_dataset(dataset_name)[dataset_split]
 
     print(f"Beginning evaluation.")
-    res = evaluate_model(transcribe_fn, ds, ds_text_column)
+    metrics, results_df = evaluate_model(transcribe_fn, ds, ds_text_column)
+
+    print(f"Evaluation done. WER={metrics.wer}, WIL={metrics.wil}.")
+    
+    file_name = f"{args.model.replace('/', '_')}_{dataset_name.replace('/', '_')}_results.csv"
```
Comment (yairl) [inline evaluate_model.py]: How about have that as an argument to the script, and also separately add the model as a column?
That way the user can decide on calling conventions themselves.

File: evaluate_model.py [hunk 2]
```diff
@@ -232,11 +235,34 @@ def evaluate_model(transcribe_fn, ds, text_column):
...
         eval_texts.append(eval_text)
 
-        res = jiwer.process_words([ref_text], [eval_text])
-        print(f"Evaluated {i+1}/{len(ds)} entries, WER={res.wer}, WIL={res.wil}")
-
-    return jiwer.process_words(ref_texts, eval_texts)
+        entry_metrics  = jiwer.process_words([ref_text], [eval_text])
+
+        entry_data = {
+            'entry_id': i,
```
Comment (yairl) [inline evaluate_model.py]: entry_id -> id?

**Review by yairl — COMMENTED** [review]
A couple of suggestions.
Let me know what you think and let's merge afterwards.

File: evaluate_model.py [hunk 3]
```diff
@@ -278,6 +305,22 @@ def evaluate_model(transcribe_fn, ds, text_column):
...
+
+    print(f"Evaluation done. WER={metrics.wer}, WIL={metrics.wil}.")
+    
+    # Add model and dataset info as columns 
+    results_df['model'] = args.model
+    results_df['dataset'] = dataset_name
+    results_df['dataset_split'] = dataset_split
+    results_df['engine'] = args.engine
+    
+    output_file = args.output if args.output else "evaluation_results"
```
Comment (yairl) [inline evaluate_model.py:318]: You can use a default argument in argparse, avoiding this code.

File: evaluate_model.py [hunk 4]
```diff
@@ -278,6 +305,22 @@ def evaluate_model(transcribe_fn, ds, text_column):
...
+    
+    results_df.to_csv(output_file, encoding='utf-8', index=False)
+    
+    with open(f"{output_file}_metrics.pkl", 'wb') as f:
```
Comment (yairl) [inline evaluate_model.py:322]: Why is pickle needed?
You have the CSV anyways, loading it is the same one-liner and it's not lots of data.

Also: https://docs.python.org/3/library/pickle.html

**Review by yairl — COMMENTED** [review]
Added a few comments - will be good if you address them in a separate PR.
Merging.

**Pull request merged**